### PR TITLE
6817009: Action.SELECTED_KEY not toggled when using key binding

### PR DESCRIPTION
**Pull request by prsadhuk**
When the Action.SELECTED_KEY property action is assigned to ToggleButton and an accelerator key binding is mapped to the action, then pressing the accelerator key binding does not toggle the toggle button.
This is because SwingUtilities.notifyAction does not fire itemStateChanged event for such action related to SELECTED_KEY.
Fix is to get the Action.SELECTED_KEY command/action and fire itemStateChanged Event via JToggleButton.setSelected() call to notify propertyChange listener of the toggled property.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-6817009](https://bugs.openjdk.org/browse/JDK-6817009): Action.SELECTED_KEY not toggled when using key binding

### Reviewers
 * [Tejesh R](https://openjdk.org/census#tr) (@TejeshR13 - Committer) ⚠️ Review applies to [4219c5c7](https://git.openjdk.org/jdk/pull/12253/files/4219c5c7a00f9b85cf72c393b3a4be943a142f3a)
 * [Jayathirth D V](https://openjdk.org/census#jdv) (@jayathirthrao - **Reviewer**) ⚠️ Review applies to [4219c5c7](https://git.openjdk.org/jdk/pull/12253/files/4219c5c7a00f9b85cf72c393b3a4be943a142f3a)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/12253/head:pull/12253` \
`$ git checkout pull/12253`

Update a local copy of the PR: \
`$ git checkout pull/12253` \
`$ git pull https://git.openjdk.org/jdk.git pull/12253/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12253`

View PR using the GUI difftool: \
`$ git pr show -t 12253`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12253.diff">https://git.openjdk.org/jdk/pull/12253.diff</a>

</details>

### CODE REVIEW

File: src/java.desktop/share/classes/javax/swing/SwingUtilities.java [hunk 1]
```diff
@@ -1810,6 +1810,13 @@ else if (!stayNull && event.getKeyChar() != KeyEvent.CHAR_UNDEFINED) {
         action.actionPerformed(new ActionEvent(sender,
                         ActionEvent.ACTION_PERFORMED, command, event.getWhen(),
                         modifiers));
+        if (event.getSource() instanceof JToggleButton) {
+            JToggleButton tb = (JToggleButton)event.getSource();
```
Comment (SWinxy) [inline src/java.desktop/share/classes/javax/swing/SwingUtilities.java:1814]: Please use a pattern variable here:
```suggestion
        if (event.getSource() instanceof JToggleButton tb) {
```

**Review by SWinxy — CHANGES_REQUESTED** [review]

File: test/jdk/javax/swing/JToggleButton/TestSelectedKey.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        AbstractAction action = new AbstractAction("Toggle me") {
+            private int count = 0;
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                label.setText("Toggle me actions " + (++count));
+            }
+        };
+        action.addPropertyChangeListener(new PropertyChangeListener() {
+            public void propertyChange(PropertyChangeEvent evt) {
+                if(Action.SELECTED_KEY.equals(evt.getPropertyName())) {
```
Comment (turbanoff) [inline test/jdk/javax/swing/JToggleButton/TestSelectedKey.java:96]:
```suggestion
                if (Action.SELECTED_KEY.equals(evt.getPropertyName())) {
```

File: test/jdk/javax/swing/JToggleButton/TestSelectedKey.java [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/* @test
+   @bug 6817009
+   @summary Verifies if AAction.SELECTED_KEY not toggled when using key binding
```
Comment (jayathirthrao) [inline test/jdk/javax/swing/JToggleButton/TestSelectedKey.java:26]: This is a headful test, we need to add @headful tag.
Also summary needs to be updated to something like "Verifies that AAction.SELECTED_KEY is not toggled when key binding is used"

**Review by jayathirthrao — APPROVED** [review]
Apart from jtreg test comment, other changes looks good.
Verified regression test also.

File: test/jdk/javax/swing/JToggleButton/TestSelectedKey.java (hunk 3 above)
Comment (prsadhuk) [reply test/jdk/javax/swing/JToggleButton/TestSelectedKey.java]: ok

**Pull request closed without merging**